docs/development-api/github.md

```diff
@@ -0,0 +1,38 @@
+---
+parent: Development API
+---
+
+# GitHub
+
+---
+
+## createCheckRun
+
+Asynchronously creates a GitHub Check Run with annotations
+
+If the following environment variables are not present,
+the function will silently return without attempting to create a check run:
+
+- `BUILDKITE_REPO`
+- `BUILDKITE_COMMIT`
+- `BUILDKITE_BUILD_NUMBER`
+- `GITHUB_API_TOKEN`
+
```

Comment:
It would also be good to document the maximum number of annotations and the behaviour of the function when `annotations` exceeds that.